docroot/modules/custom/mass_media/src/Traits/MediaModerationStateActionTrait.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\mass_media\Traits;
+
+use Drupal\media\MediaInterface;
+
+/**
+ * Trait containing reusable code for media moderation state actions.
+ */
+trait MediaModerationStateActionTrait {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createRevision(MediaInterface $entity = NULL, $moderation_state) {
+    $revision_message = $this->t('Bulk action: Moderation state set to @moderation_state for media item @entity_id.', [
+      '@moderation_state' => $moderation_state,
+      '@entity_id' => $entity->id(),
```

Comment:
I think these should be passed as context variables in the `notice` call, avoiding the need to call `t()` here? That way we can ensure that if a future moderation state contains non-HTML printable characters that they will be escaped correctly.

Reply:
My primary motivation for using `t` here was to have the same message for both the revision log and the `notice` call, and `setRevisionLogMessage` doesn't support string translation as far as I can tell. With that said, I don't feel too strongly about having two different messages, especially if you think relying on the `notice` call's string translation is more beneficial.